snuba/datasets/errors_replacer.py

```diff
@@ -148,89 +157,161 @@ def get_count_query(self, table_name: str) -> Optional[str]:
 ) -> Optional[bool]:
-    return redis_client.set(
-        get_project_needs_final_key(project_id, state_name),
-        True,
-        ex=settings.REPLACER_KEY_TTL,
-    )
+    key, type_key = get_project_needs_final_key_and_type_key(project_id, state_name)
+    p = redis_client.pipeline()
+    p.set(key, True, ex=settings.REPLACER_KEY_TTL)
+    p.set(type_key, replacement_type, ex=settings.REPLACER_KEY_TTL)
+    [final_was_set, _] = p.execute()
+    return bool(final_was_set)
 
 
 def get_projects_query_flags(
     project_ids: Sequence[int], state_name: Optional[ReplacerState]
-) -> Tuple[bool, Sequence[int]]:
+) -> Tuple[bool, Sequence[int], Set[str]]:
     """\
     1. Fetch `needs_final` for each Project
     2. Fetch groups to exclude for each Project
     3. Trim groups to exclude ZSET for each Project
+    4. Fetch replacement types for each Project
 
-    Returns (needs_final, group_ids_to_exclude)
+    Returns (needs_final, group_ids_to_exclude, replacement_types)
     """
 
     s_project_ids = set(project_ids)
     now = time.time()
     p = redis_client.pipeline()
 
-    needs_final_keys = [
-        get_project_needs_final_key(project_id, state_name)
+    needs_final_keys_and_type_keys = [
+        get_project_needs_final_key_and_type_key(project_id, state_name)
         for project_id in s_project_ids
     ]
-    for needs_final_key in needs_final_keys:
+
+    for needs_final_key, _ in needs_final_keys_and_type_keys:
         p.get(needs_final_key)
 
-    exclude_groups_keys = [
-        get_project_exclude_groups_key(project_id, state_name)
+    exclude_groups_keys_and_types = [
+        get_project_exclude_groups_key_and_type_key(project_id, state_name)
         for project_id in s_project_ids
     ]
-    for exclude_groups_key in exclude_groups_keys:
+
+    for exclude_groups_key, _ in exclude_groups_keys_and_types:
         p.zremrangebyscore(
             exclude_groups_key, float("-inf"), now - settings.REPLACER_KEY_TTL
         )
         p.zrevrangebyscore(
             exclude_groups_key, float("inf"), now - settings.REPLACER_KEY_TTL
         )
 
+    for _, needs_final_type_key in needs_final_keys_and_type_keys:
+        p.get(needs_final_type_key)
+
+    for _, type_key in exclude_groups_keys_and_types:
+        p.zremrangebyscore(type_key, float("-inf"), now - settings.REPLACER_KEY_TTL)
+        p.zrevrangebyscore(type_key, float("inf"), now - settings.REPLACER_KEY_TTL)
+
     results = p.execute()
 
-    needs_final = any(results[: len(s_project_ids)])
+    return process_exclude_groups_and_replacement_types_results(
+        results, len(s_project_ids)
+    )
+
+
+def process_exclude_groups_and_replacement_types_results(
+    results: List[Any], len_projects: int
+) -> Tuple[bool, Sequence[int], Set[str]]:
+    """
+    Helper function for `get_projects_query_flags`.
+    `results` is in the form:
+    [
+        needs_final...,
+        exclude_groups...,
+        project_replacement_types...,
```

Comment:
you may want to rename `project_replacement_types` to `needs_final_replacement_types`. A project replacement is not something.